allauth_2fa/forms.py

```diff
@@ -62,9 +65,13 @@ def __init__(self, user, **kwargs):
 
     def save(self):
         # Delete any backup tokens.
-        static_device = self.user.staticdevice_set.get(name='backup')
-        static_device.token_set.all().delete()
-        static_device.delete()
+        try:
+            static_device = self.user.staticdevice_set.get(name=app_settings.BACKUP_DEVICE_NAME)
+        except ObjectDoesNotExist:
+            pass
+        else:
+            static_device.token_set.all().delete()
```

Comment:
Looking at this code more, I don't think this line is necessary, deleting `static_device` should cascade and delete the tokens.

Reply:
This is true.